### DISCUSSION TITLE
How to hard delete objects older than n-days in LakeFS?

### DISCUSSION THREAD

**Discussion by nikolaykalachev** [treeverse/lakeFS#2761] [Q&A]
Guuuuuys, Hi 👋,
How to hard delete objects older than n-days in LakeFS?

**Comment by nikolaykalachev** [comment]
It would be a scheduled job.
I need to find and delete old files.

**Reply by arielshaqed** [reply]
Hi @nikolaykalachev,
Does https://docs.lakefs.io/reference/garbage-collection.html help?

**Reply by nikolaykalachev** [reply]
> Hi @nikolaykalachev, Does https://docs.lakefs.io/reference/garbage-collection.html help?

bsaically, yes. but before i need to softdelete the old files. so i don't know is there is any native config for that or i need to use a handmade script on python or spark (and wich one is simpler)

**Reply by nikolaykalachev** [reply]
i mean a bulk delete
